src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs

```diff
@@ -517,21 +514,28 @@ public object Dequeue(ThreadPoolWorkQueueThreadLocals tl, ref bool missedSteal)
             return callback;
         }
 
-        internal static bool Dispatch()
+        /// <summary>
+        /// Dispatches work items to this thread.
+        /// </summary>
+        /// <returns>
+        /// <c>true</c> if this thread did as much work as was available or its quantum expired.
+        /// <c>false</c> if this thread stopped working early.
+        /// </returns>
+        internal static bool Dispatch(bool failFastOnExceptions = true)
```

Comment:
I do not think we need `failFastOnExceptions = true` argument. The runtimes that need to handle exceptions escaping from the threadpool can add `try / catch` around the `Dispatch` call.

Reply:
That was my other option. I want to move the CoreRT "ThreadPool.Portable" implementation as an optional feature to shared code base next (and use it from Mono). It made sense to keep the try-catch in the common code, but I can move it to separate method.

Reply:
I think having the try/catch in the runtime-specific entrypoint would be better.

Reply:
Done